### ISSUE TITLE
Random crash on Win 7

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#29]
I've recently begun using Mixxx 1.8.0 (final) on a Windows 7 desktop. It's a completely fresh install (deleted anything related to Mixxx before installing, cleaned registry, etc.) and I'm only able to use it for a few minutes before it closes. I don't receive any error message or even the Windows dialog saying an application quit unexpectedly or has stopped responding or anything. It just closes. I'm not entirely sure how you would go about reproducing the problem unless this is pretty common...

PC:
Windows 7 Home Premium (32-bit) [all recent updates installed]
2.4GHz Athlon 64 (AMD Athlon(tm) 64 X2 Dual Core Processor 4400+)
GeForce 6150SE nForce 430 video card (128MB, using most recent driver)
Realtek High Definition Audio sound card

**Comment by Holzhaus** [comment]
(by rryan)
Hi Josh,

Thanks for taking the time to file a bug. Could you please tell me what settings you have in your in your Audio preferences, specifically the Audio API, the Master/Headphone devices you chose, your latency, and your sample rate? 

Are you using any MIDI controller or Vinyl Control?

Thanks,
RJ Ryan

**Comment by Holzhaus** [comment]
(by jpv41193)
Nope, no MIDI or Vinyl Control, just some real simple mixing (I'm just getting into DJing)

My latency is set to 42ms, my master device is just "primary sound driver", no headphones, sample rate is 48,000, audio API is Windows Direct Sound.

**Comment by Holzhaus** [comment]
(by gamegod)
Hi Josh,

Regarding the crash, do you typically have vinyl emulation or pitch
independent time stretch selected in the preferences?

Can you please see if you can reproduce the crash if you switch to the other
option?

Thanks,
Albert
On 2010-10-09 9:05 PM, "Josh Vickerson" <email address hidden> wrote:
> I tried increasing the latency to 85ms with my original settings and had
the same result.
> I've tried 44.1kHz (46s latency) and got the same result.
>
> I can't use WASAPI because the only output I can select then is the
> digital out (which I have no speakers for).
>
> I'm not brand new to Mixxx but I'm not terribly experienced. I used
> 1.7.2 on this same computer for a few weeks before 1.8.0 was released
> and had no issues (and was able to set my latency to 21ms without
> problems). All in all I've noticed a slight decrease in performance from
> this release. I tried using 1.7.2 again and started looking for more
> differences. In 1.8.0 there is a slight skip (I mean like a fraction of
> a second) whenever a song is playing and I load another into the other
> track. If I skip to a point in a song by clicking the wave-form while
> it's playing it skips a little bit (both of these happened with 85ms
> latency). I had neither of these issues with 1.7.2.
>
> I've downloaded the ASIO API but haven't gotten to trying it yet.
> Thought I'd give you an update in the meantime.
>
> --
> Random crash on Win 7
> https://bugs.launchpad.net/bugs/657172
> You received this bug notification because you are a member of Mixxx
> Development Team, which is subscribed to Mixxx.

**Comment by Holzhaus** [comment]
(by jpv41193)
So using my original settings I described and changing from vinyl emulation to independent time stretch it doesn't seem to crash. Before I would use it for about 20-30 minutes and then it would just close. I was able to use it as I normally would for 45 minutes without it crashing by selecting independent time stretch.

**Comment by Holzhaus** [comment]
(by jpv41193)
To be clear, I only stopped using it after 45 minutes, it never crashed.

**Comment by Holzhaus** [comment]
(by rryan)
Hmm, ok thanks. Could you post your mixxx.log file? You should be able to find this in C:\Program Files\Mixxx\mixxx.log. I'm not totally familiar with our Windows installer so it might be slightly different.

**Comment by Holzhaus** [comment]
(by rryan)
(btw I meant post your mixxx.log right after it crashes)

**Comment by Holzhaus** [comment]
(by gamegod)
I found a bug in EBSL using Valgrind that is likely the cause of this crash. I've added a workaround that clamps prev_sample in enginebufferscale.cpp to prevent the out-of-bounds access, since we're probably going to be doing something drastic with this code in the near future. 

The fix is committed in the 1.8 branch, r2653. 

Josh: Thanks for reporting this bug. A crash like this is a critical bug and fortunately you reported it so we were able to fix it sooner rather than later. We will be releasing 1.8.1 as soon as we can to get the fix out.
